tokio/src/task/coop/mod.rs

```diff
@@ -275,20 +280,63 @@ cfg_coop! {
         }
     }
 
-    /// Returns `Poll::Pending` if the current task has exceeded its budget and should yield.
+    /// Decrements the task budget and returns `Poll::Pending` if the budget is depleted.
+    /// This indicates that the task should yield to the scheduler. Otherwise, returns
+    /// `RestoreOnPending` which can be used to commit the budget consumption.
     ///
-    /// When you call this method, the current budget is decremented. However, to ensure that
-    /// progress is made every time a task is polled, the budget is automatically restored to its
-    /// former value if the returned `RestoreOnPending` is dropped. It is the caller's
-    /// responsibility to call `RestoreOnPending::made_progress` if it made progress, to ensure
-    /// that the budget empties appropriately.
+    /// The returned [`RestoreOnPending`](RestoreOnPending) will revert the budget to its former
+    /// value when dropped unless [`RestoreOnPending::made_progress`](RestoreOnPending::made_progress)
+    /// is called. It is the caller's responsibility to do so when it _was_ able to
+    /// make progress after the call to `poll_pending`.
+    /// Restoring the budget automatically ensures the task can try to make progress in some other
+    /// way.
     ///
     /// Note that `RestoreOnPending` restores the budget **as it was before `poll_proceed`**.
     /// Therefore, if the budget is _further_ adjusted between when `poll_proceed` returns and
-    /// `RestRestoreOnPending` is dropped, those adjustments are erased unless the caller indicates
+    /// `RestoreOnPending` is dropped, those adjustments are erased unless the caller indicates
     /// that progress was made.
+    ///
+    /// # Examples
+    ///
+    /// This example shows a `Future` implementation that uses `poll_proceed` to participate in
+    /// cooperative scheduling. If the work to be done consists of polling another future, consider
+    /// using the higher-level [`cooperative`](cooperative) function instead.
+    ///
+    /// ```ignore
```

Comment:
We generally try pretty hard to avoid `ignore` on doctests. Please try to make the doc-test compile.

Reply:
Ok, the issue was the `...` blank I left for whatever user code would be there. Not sure what a good, not overly specific, example might be that's not just polling a Future.

Reply:
🙄 example would be great if it actually worked. That'll teach me not to try and write working code directly in doc comments.

Reply:
Fixed the code. In the end I think the concrete example is better than the generic one. Maybe the code is a bit too naive for this codebase, but for documentation it's probably best to keep it simple.

Reply:
Simple is good.